src/multilevel.jl

```diff
@@ -4,20 +4,62 @@ struct Level{T,V}
     R::SparseMatrixCSC{T,V}
 end
 
-struct MultiLevel{S, Pre, Post, Ti, Tv}
+struct MultiLevel{S, Pre, Post, Ti, Tv, TW}
     levels::Vector{Level{Ti,Tv}}
     final_A::SparseMatrixCSC{Ti,Tv}
     coarse_solver::S
     presmoother::Pre
     postsmoother::Post
+    workspace::TW
```

Comment:
I like the fact that we now store this debugging information. Is there a way we can not store it by default and give the user the option to store it?

Reply:
This is not for debugging, it is for pre-allocation. This is how we can do a fully inplace solve, so the `ldiv!` now has 0 allocations.